eth_tester/utils/backend_testing.py

```diff
@@ -474,12 +486,112 @@ def test_estimate_gas(self, eth_tester):
             math_address,
             'increment',
         )
-        gas_estimation = eth_tester.estimate_gas(dissoc(estimate_call_math_transaction, 'gas'))
+        if not is_pyevm_available():  # py-evm requires the gas parameter
+            estimate_call_math_transaction = dissoc(estimate_call_math_transaction, 'gas')
+        gas_estimation = eth_tester.estimate_gas(estimate_call_math_transaction)
         call_math_transaction = assoc(estimate_call_math_transaction, 'gas', gas_estimation)
         transaction_hash = eth_tester.send_transaction(call_math_transaction)
         receipt = eth_tester.get_transaction_receipt(transaction_hash)
         assert receipt['gas_used'] == gas_estimation
 
+    def test_can_call_after_exception_raised_calling_pyethereum(self, eth_tester):
```

Comment:
The idea behind this module is that you should be able to write the test once, and it tests across all backends. As much as possible, the caller should be agnostic to which backend is being used.

Anything that doesn't have a unified interface would be a bug.